hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -235,6 +236,7 @@ public void handleRound(
             @NonNull final Round round,
             @NonNull final Consumer<ScopedSystemTransaction<StateSignatureTransaction>> stateSignatureTxnCallback) {
         logStartRound(round);
+        BlockBufferService.ensureNewBlocksPermitted();
```

Comment:
Can we guard it here and in `BlockStreamManagerImpl`, just in two places instead of all methods in `BlockBufferService` with `isStreamingEnabled` ?